src/context/meta-tags-context.php

```diff
@@ -621,9 +621,10 @@ public function generate_main_image_id() {
 			case is_search():
 			case is_date():
 			case is_post_type_archive():
-				if ( ! empty ( $GLOBALS['wp_query']->posts ) ) {
```

Comment:
This change has been added to this PR since it was not possible to pull from trunk (or the release branch), where the change had been made.